frontend/src/pages/modelServing/screens/projects/ModelServingPlatform.tsx

```diff
@@ -167,7 +167,7 @@ const ModelServingPlatform: React.FC = () => {
         }
         isLoading={!servingRuntimesLoaded && !templatesLoaded}
         isEmpty={shouldShowPlatformSelection}
-        loadError={platformError || servingRuntimeError || templateError}
+        loadError={platformError || servingRuntimeError || (templateError as Error | undefined)}
```

Comment:
The cast is not ideal. I forget why they type error as `unknown` instead of some super of `Error` union with `undefined`.
@andrewballantyne any recollection here?

Reply:
No recollection -- anyway we can wrap and override that instead of casting everywhere?

There was some gaps in the K8s CRUD usages too -- I think it was just short sightedness that had not been vetted.

Reply:
Agreed. We can create our own hook that can perform a typeof check on the error and if it's of type error use it, otherwise create a new Error object. Then we use this hook everywhere in place of `useK8sWatchResource`

Reply:
yes, I will work on this.